packages/vue-app/template/mixins/fetch.server.js

```diff
@@ -0,0 +1,47 @@
+import Vue from 'vue'
+import { hasFetch, normalizeError } from '../utils'
+
+function getDataDiff(o1, o2) {
+  return Object.keys(o2).reduce((diff, key) => {
+    if (o1[key] === o2[key]) {
+      return diff
+    }
+    return {
+      ...diff,
+      [key]: o2[key]
+    }
+  }, {})
+}
+
+export default {
+  beforeCreate () {
+    if (hasFetch(this)) {
+      this._fetchOnServer = this.$options.fetchOnServer !== false
+      Vue.util.defineReactive(this, '$fetchState', {
+        pending: !this._fetchOnServer,
+        error: null,
+        timestamp: Date.now()
+      })
+    }
+  },
+  async serverPrefetch () {
```

Comment:
Not sure if this is also possible to add programmatically, if not would it make sense to cache the hasFetch result in beforeCreate? (Probably not)

Reply:
@pimlie I refactored to use cache in beforeCreate. Let's keep programmatically conv at https://github.com/nuxt/nuxt.js/pull/6880#discussion_r372854230